sos/report/plugins/smclient.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin
+
+
+class SMcli(Plugin, IndependentPlugin):
+
+    short_desc = 'SANtricity storage device'
+
+    plugin_name = 'smclient'
+    plugin_timeout = 900
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('SMclient',)
+
+    option_list = [
+        ("debug", "capture support debug data", "slow", False),
+    ]
+
+    def setup(self):
+        subcmds = [
+            "show storagearray;",
+            "show storagearray connections;",
+            "show storagearray healthstatus;",
+        ]
+
+        ssnames = []
+
+        # Get list of storage arrays
+        result = self.collect_cmd_output('SMcli -d -S')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if 'localhost' in line:
+                    ssnames.append(line.split()[0])
+
+        # Collect status of each storage array
+        for ssname in ssnames:
+            self.add_cmd_output([
+                "SMcli localhost -n %s -c '%s'" % (ssname, subcmd)
+                for subcmd in subcmds
+            ])
+
+        if self.get_option("debug"):
+            self.do_debug(ssnames)
+
+    def do_debug(self, ssnames):
+        subcmds = [
+            "save storageArray supportData file=\"supportbundle\";",
+        ]
+
+        logpath = self.get_cmd_output_path(make=False)
+
+        for ssname in ssnames:
+            self.add_cmd_output([
+                "SMcli localhost -n %s -c '%s'" % (ssname, subcmd)
+                for subcmd in subcmds
+            ], runat=logpath, timeout=450)
```

Comment:
I'm assuming this generates some file in the working directory, and that the filename would be `supportbundle`?

If so, my first question would be is "supportbundle" the term used by the team(s)/user(s) that look at this data?

Secondly, if there are more than 1 entities in `ssnames`, would this not overwrite the `supportbundle` file, or is `SMcli` appending this data to that file if it already exists?

Reply:
I have updated the code to generate a unique filename. It is going to look like `support-myhost_rbod_nbj089.7z`. The file is generated by the SMcli command in the current working directory. Hence, the `logat` is needed.

As you said, the user will use the file to look at the data to troubleshooting issues.

Thank you for the feedback.